### DISCUSSION TITLE
Interpreting the common domain model

### DISCUSSION THREAD

**Discussion by Teyiowuawi** [finos/common-domain-model#3172] [General]
I work with a team of data modellers and we wanted to further understand the common domain model. 

How would this class and its condition (especially the condition one-of) be interpreted? My understanding is that the condition indicates the unity type must be one of (capacity unit, weather unit, financial unit or currency) Is this correct?

type UnitType: <"Defines the unit to be used for price, quantity, or other purposes">
    capacityUnit CapacityUnitEnum (0..1) <"Provides an enumerated value for a capacity unit, generally used in the context of defining quantities for commodities.">
    weatherUnit WeatherUnitEnum (0..1) <"Provides an enumerated values for a weather unit, generally used in the context of defining quantities for commodities.">
    financialUnit FinancialUnitEnum (0..1) <"Provides an enumerated value for financial units, generally used in the context of defining quantities for securities.">
    currency string (0..1) <"Defines the currency to be used as a unit for a price, quantity, or other purpose.">
        [metadata scheme]

    condition UnitType: <"Requires that a unit type must be set.">
        one-of

**Comment by chrisisla** [comment]
Hi @Teyiowuawi 
Yes that is correct, the `one-of` expression is stating that one of the attributes in the class must be included.

For more details on conditions in general this is a useful link:
https://docs.rosetta-technology.io/rosetta/rosetta-dsl/rune-modelling-component/#condition

And this has more details on `one-of`:
https://docs.rosetta-technology.io/rosetta/rosetta-dsl/rune-modelling-component/#one-of

Thanks!
